src/lib.rs

```diff
@@ -1449,6 +1450,13 @@ pub trait Itertools : Iterator {
         adaptors::tuple_combinations(self)
     }
 
+    fn array_combinations<const R: usize>(self) -> ArrayCombinations<Self::Item, R>
+        where Self: Sized,
+              Self::Item: Clone + std::fmt::Debug,
+    {
+        ArrayCombinations::new(self.collect())
```

Comment:
Iterator adapters should advance the underlying iterator lazy, not `collect`ing them when they're created. If you look at the implementation of `Combinations` that I linked in the other pr you'll see that it stores a `LazyBuffer` that takes care of advancing the underlying iterator only when needed.

Reply:
I was thinking about that, I'll try it. Hopefully it won't cause too much slow downs